src/material/menu/menu.ts

```diff
@@ -320,13 +320,35 @@ export class _MatMenuBase implements AfterContentInit, MatMenuPanel<MatMenuItem>
    * @param origin Action from which the focus originated. Used to set the correct styling.
    */
   focusFirstItem(origin: FocusOrigin = 'program'): void {
+    const manager = this._keyManager;
+
     // When the content is rendered lazily, it takes a bit before the items are inside the DOM.
     if (this.lazyContent) {
       this._ngZone.onStable.asObservable()
         .pipe(take(1))
-        .subscribe(() => this._keyManager.setFocusOrigin(origin).setFirstItemActive());
+        .subscribe(() => manager.setFocusOrigin(origin).setFirstItemActive());
     } else {
-      this._keyManager.setFocusOrigin(origin).setFirstItemActive();
+      manager.setFocusOrigin(origin).setFirstItemActive();
+    }
+
+    // If there's no active item at this point, it means that all the items are disabled.
+    // Move focus to the menu panel so keyboard events like Escape still work. Also this will
+    // give _some_ feedback to screen readers.
+    if (!manager.activeItem && this._directDescendantItems.length) {
+      let element = this._directDescendantItems.first._getHostElement().parentElement;
+
+      // Because the `mat-menu` is at the DOM insertion point, not inside the overlay, we don't
+      // have a nice way of getting a hold of the menu panel. We can't use a `ViewChild` either
+      // because the panel is inside an `ng-template`. We work around it by starting from one of
+      // the items and walking up the DOM.
+      while (element) {
+        if (element.getAttribute('role') === 'menu') {
+          element.focus();
```

Comment:
Note: it felt a little weird to have this logic here since the method is called `focusFirstItem`, however the alternative is to do it inside the menu trigger which would require another method to be added to the `MatMenuPanel` interface which will increase the API surface. Furthermore, doing it through the trigger may no be very reliable, because we'd need to check if the `document.activeElement` is inside the menu and some browsers move focus asynchronously.